### PR TITLE
app-text/scdoc: new ebuild

### PR DESCRIPTION
**Pull request by Hummer12007**
initial version 1.2.3

### CODE REVIEW

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+src_compile() {
+	if tc-is-cross-compiler; then
+		tc-export_build_env
+		MAKEOPTS+=" HOST_SCDOC=${S}/hostscdoc"
+		emake scdoc OUTDIR="${S}/.build.host" CC=$(tc-getBUILD_CC) \
+			CFLAGS="${BUILD_CFLAGS}" LDFLAGS="${BUILD_LDFLAGS}" ||
+				die "Failed to build host scdoc"
+		mv scdoc hostscdoc
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: `|| die`

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		die 'Failed to patch makefile'
+
+	default
+}
+
+src_compile() {
+	if tc-is-cross-compiler; then
+		tc-export_build_env
+		MAKEOPTS+=" HOST_SCDOC=${S}/hostscdoc"
+		emake scdoc OUTDIR="${S}/.build.host" CC=$(tc-getBUILD_CC) \
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: no `|| die` after emake for `EAPI >= 4`

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+DESCRIPTION="Standalone tool for generating man pages with a simple syntax"
+HOMEPAGE="https://git.sr.ht/~sircmpwn/scdoc"
+SRC_URI="https://git.sr.ht/~sircmpwn/scdoc/snapshot/${P}.tar.xz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: do you really need this empty?

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+RDEPEND=""
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: and this too?

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+src_compile() {
+	if tc-is-cross-compiler; then
+		tc-export_build_env
+		MAKEOPTS+=" HOST_SCDOC=${S}/hostscdoc"
+		emake scdoc OUTDIR="${S}/.build.host" CC=$(tc-getBUILD_CC) \
+			CFLAGS="${BUILD_CFLAGS}" LDFLAGS="${BUILD_LDFLAGS}" ||
+				die "Failed to build host scdoc"
+		mv scdoc hostscdoc
+	fi
+	emake LDFLAGS="${LDFLAGS}" || die 'Make failed'
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: Again, no `|| die` needed

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
+RDEPEND=""
+
+src_prepare() {
+	sed -i 's/-Werror//' Makefile ||
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: why in two lines? (minor one, though)

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		die 'Failed to remove -Werror'
+
+	sed -i 's/CFLAGS=/CFLAGS+=/' Makefile ||
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: the same

**Comment by Hummer12007** [comment]
Thanks, fixed!

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	default
+}
+
+src_compile() {
+	if tc-is-cross-compiler; then
+		tc-export_build_env
+		MAKEOPTS+=" HOST_SCDOC=./hostscdoc"
+		emake scdoc OUTDIR="${S}/.build.host" CC=$(tc-getBUILD_CC) \
+			CFLAGS="${BUILD_CFLAGS}" LDFLAGS="${BUILD_LDFLAGS}"
+		mv scdoc hostscdoc
```
Comment (Zlogene) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: still `|| die` after `mv` command

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+src_prepare() {
+	sed -i 's/-Werror//' Makefile || die 'Failed to remove -Werror'
+
+	sed -i 's/CFLAGS=/CFLAGS+=/' Makefile || die 'Failed to patch makefile'
+
+	default
```
Comment (SoapGentoo) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: can't you replace the `sed`'s with a patch?

File: app-text/scdoc/scdoc-1.2.3.ebuild [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+src_compile() {
+	if tc-is-cross-compiler; then
+		tc-export_build_env
+		MAKEOPTS+=" HOST_SCDOC=./hostscdoc"
```
Comment (SoapGentoo) [inline app-text/scdoc/scdoc-1.2.3.ebuild]: mucking with `MAKEOPTS` isn't allowed (I know this is common, still not allowed), append it to the `emake` line

File: app-text/scdoc/scdoc-1.2.3.ebuild (hunk 9 above)
Comment (Hummer12007) [reply app-text/scdoc/scdoc-1.2.3.ebuild]: It's less likely to require changes than a patch in future (and easier to keep note of for me).

**Comment by Hummer12007** [comment]
Another round of fixes.

File: app-text/scdoc/scdoc-1.2.3.ebuild (hunk 9 above)
Comment (SoapGentoo) [reply app-text/scdoc/scdoc-1.2.3.ebuild]: then please do them after `default`

**Pull request closed without merging**